pro/src/screens/OfferIndividual/StocksEventCreation/form/validationSchema.ts

```diff
@@ -32,14 +32,46 @@ export const getValidationSchema = (priceCategoriesOptions: SelectOption[]) =>
     beginningTimes: yup
       .array()
       .of(yup.string().nullable().required('Veuillez renseigner un horaire')),
-    quantityPerPriceCategories: yup.array().of(
-      yup.object().shape({
-        quantity: yup.number(),
-        priceCategory: oneOfSelectOption(
-          yup.string().required('Veuillez renseigner un tarif'),
-          priceCategoriesOptions
-        ),
-      })
-    ),
+    quantityPerPriceCategories: yup
+      .array()
+      .of(
+        yup.object().shape({
+          quantity: yup.number(),
+          priceCategory: oneOfSelectOption(
+            yup.string().required('Veuillez renseigner un tarif'),
+            priceCategoriesOptions
+          ),
+        })
+      )
+      .test('isPriceCategoryUnique', function (list) {
+        if (!list) return
+        const price_category_map = list.map(a => a.priceCategory)
+        const duplicateIndex = price_category_map.findIndex(
+          (item, index) => price_category_map.lastIndexOf(item) !== index
+        )
+
+        if (duplicateIndex != -1) {
+          const errorsIndex = price_category_map.reduce<number[]>(function (
+            accumulator,
+            currentValue,
+            currentIndex
+          ) {
+            if (currentValue === price_category_map[duplicateIndex])
+              accumulator.push(currentIndex)
+            return accumulator
+          },
+          [])
```

Comment:
J'ai l'impression que tu pourrais créer cette liste à la place de `duplicateIndex` ligne 49, et vérifier si la liste est vide ou pas ligne 53.

Reply:
De ma compréhension du code, je pense qu'on pourrait remplacer tout le test par : 

```suggestion
 .test('isPriceCategoryUnique', function (list) {
        if (!list) return
        const price_category_map = list.map(a => a.priceCategory)
        const isPriceCategoryUnique =
          new Set(price_category_map).size === price_category_map.length
        return isPriceCategoryUnique
      }),
```